service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -3,27 +3,46 @@
 import com.epam.esm.converter.DTOConverter;
 import com.epam.esm.dto.PageDTO;
 import com.epam.esm.dto.PageRequestDTO;
-import com.epam.esm.dto.UserDTO;
+import com.epam.esm.dto.UserRequestFieldDTO;
+import com.epam.esm.dto.entityDTO.UserDTO;
+import com.epam.esm.entity.Role;
 import com.epam.esm.entity.User;
+import com.epam.esm.entity.UserRole;
+import com.epam.esm.exception.EntityAlreadyExistsException;
 import com.epam.esm.exception.EntityNotFoundException;
 import com.epam.esm.repository.Repository;
+import com.epam.esm.repository.RoleRepositoryImpl;
+import com.epam.esm.repository.UserRepository;
 import com.epam.esm.service.AbstractService;
 import com.epam.esm.service.UserService;
 import com.epam.esm.specification.CriteriaSpecification;
 import com.epam.esm.specification.user.UserAllSpecification;
 import com.epam.esm.util.PageRequestDTOHandler;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 @Service
 public class UserServiceImpl extends AbstractService<UserDTO, User> implements UserService {
-
+    private final UserRepository userRepository;
```

Comment:
You don't need `private final UserRepository userRepository` field in this class at all. You already have UserRepository in super class.

Reply:
I did this in order to use specific methods of UserRepository in the service: userRepository.findByEmail(userDTO.getEmail())